[PRD-999] add nodes internal addresses into hosts file for name resoluti...

aglarendil opened the pull request:
...on

aglarendil committed:
[PRD-999] add nodes internal addresses into hosts file for name resolution

mirantis-jenkins commented:
:+1: Test PASSed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/108/

dmitryilyin commented:
There is a good built it Puppet resource for /etc/hosts content management.

``` ruby
Host {
  ensure => present,
}

host { 'controller01.domain.ltd' :
  ip => '192.168.0.2',
  host_aliases => [ 'c1', 'controller1', 'controller01' ],
  comment => 'The first Fuel controller',
}
```

It doesn't forbid you from manual editing of /etc/hosts too, but will enforce attributes of described hosts without touching unmanaged ones.

You can also list all installed host resources
**puppet resource host**

Get help about host resource type
**puppet describe host**

And get information about specific host
**puppet resource host localhost**

``` ruby
host { 'localhost':
  ensure => 'present',
  ip     => '127.0.0.1',
  target => '/etc/hosts',
}
```

There are not so many things Puppet is good for but this is definitly one of them)

http://docs.puppetlabs.com/references/latest/type.html#host

aglarendil committed:
[PRD-999] Add custom function and use Puppet Host

Add nodes_to_hosts converter function
Rewrite manifest to use Puppet builtin Host type
Trigger create_resources for Host generation

aglarendil committed:
[PRD-999] remove unneeded template

dmitryilyin commented:
Great and elegant solution! +1

mirantis-jenkins commented:
:+1: Test PASSed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/178/

mirantis-jenkins commented:
:+1: Test PASSed.
Refer to this link for build results: http://fuel-jenkins.mirantis.com/job/fuel_syntax_check/179/
